### PR TITLE
ACT-1296: Infura Faucet Integration in MetaMask Docs

### PR DESCRIPTION
**Pull request by Voltmod**
[ACT-1296](https://infura.atlassian.net/browse/ACT-1296) Infura Faucet Integration in MetaMask Docs

### CODE REVIEW

**Comment by Chator1** [comment]
Metamask:0x84e8277c938102f42FC2D9625Fc75049A86c99Ae
Abdulhafez 
bdalhafzslah382@gmail.com

File: src/components/NavbarWallet/index.tsx [hunk 1]
```diff
@@ -83,7 +83,7 @@ const NavbarWalletComponent: FC = ({
 const NavbarWallet = (props) => {
```
Comment (Chator1) [inline src/components/NavbarWallet/index.tsx:83]: 0x84e8277c938102f42FC2D9625Fc75049A86c99Ae

**Review by aednikanov — APPROVED** [review]

**Review by TrofimovAnton85 — DISMISSED** [review]

**Review by TrofimovAnton85 — APPROVED** [review]

File: src/components/Faucet/Faq.tsx [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+              <span>
+                Have ideas on how to improve the faucet? Awesome! We’d love to
+                hear them. Submit them{" "}
+                <a
+                  data-testid="faq-linea-discord"
+                  onClick={handleClickDiscord}
+                  target="_blank"
+                  href="https://discord.com/invite/consensys"
+                >
+                  here.
```
Comment (alexandratran) [inline src/components/Faucet/Faq.tsx:160]:
```suggestion
                If you have ideas on how to improve the faucet,
                feel free to submit them on{" "}
                <a
                  data-testid="faq-linea-discord"
                  onClick={handleClickDiscord}
                  target="_blank"
                  href="https://discord.com/invite/consensys"
                >
                  Discord.
```

File: src/components/Faucet/Faq.tsx [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+              </span>
+            </Text>
+          </Accordion>
+          <Accordion>
+            <Text as="h3" className={styles.accordionHeader}>
+              Where does Linea ETH come from?
+            </Text>
+            <Text as="p" className={styles.accordionContainer}>
+              <span>
+                Linea ETH were intially Goerli ETH that were bridged to Linea
```
Comment (alexandratran) [inline src/components/Faucet/Faq.tsx:171]:
```suggestion
                Linea ETH comes from Goerli ETH that is bridged to Linea
```

File: src/components/Faucet/Faq.tsx [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+                Linea ETH?x
+              </Text>
+              <Text as="p" className={styles.accordionContainer}>
+                We require an address with Ethereum Mainnet activity to
+                safeguard the faucet from automated bots, ensuring equitable
+                Linea ETH distribution. The amount of Linea ETH you can receive
+                ranges from 0 to 0.5, depending on your address’s level of
+                activity. No activity results in no Linea ETH. while a higher
+                number of transactions can earn you up to 0.5. We maintain
+                confidentiality over the exact criteria used to determine the
```
Comment (alexandratran) [inline src/components/Faucet/Faq.tsx:87]:
```suggestion
                MetaMask requires an address with Ethereum Mainnet activity to
                safeguard the faucet from automated bots, ensuring equitable
                Linea ETH distribution. The amount of Linea ETH you can receive
                ranges from 0 to 0.5, depending on your address's level of
                activity. No activity results in no Linea ETH, while a higher
                number of transactions can earn you up to 0.5. MetaMask maintains
                confidentiality over the exact criteria used to determine the
```

File: src/components/Faucet/Faq.tsx [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+                Why must my address have Ethereum Mainnet activity to claim
+                Sepolia ETH?
+              </Text>
+              <Text as="p" className={styles.accordionContainer}>
+                We require an address with Ethereum Mainnet activity to
+                safeguard the faucet from automated bots, ensuring equitable
+                Sepolia ETH distribution. The amount of Sepolia ETH you can
+                receive ranges from 0 to 0.5, depending on your address’s level
+                of activity. No activity results in no Sepolia ETH. while a
+                higher number of transactions can earn you up to 0.5. We
```
Comment (alexandratran) [inline src/components/Faucet/Faq.tsx:207]:
```suggestion
                MetaMask requires an address with Ethereum Mainnet activity to
                safeguard the faucet from automated bots, ensuring equitable
                Sepolia ETH distribution. The amount of Sepolia ETH you can
                receive ranges from 0 to 0.5, depending on your address's level
                of activity. No activity results in no Sepolia ETH, while a
                higher number of transactions can earn you up to 0.5. MetaMask
```

File: src/components/Faucet/Faq.tsx [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+                number of transactions can earn you up to 0.5. We maintain
+                confidentiality over the exact criteria used to determine the
+                amount issued to prevent any exploitation of the system, aiming
+                to distribute testnet ETH fairly among genuine, active users.
+              </Text>
+            </Accordion>
+          )}
+          <Accordion>
+            <Text as="h3" className={styles.accordionHeader}>
+              I’m new to Web3. What is a faucet?
```
Comment (alexandratran) [inline src/components/Faucet/Faq.tsx:95]:
```suggestion
              I'm new to Web3. What is a faucet?
```

File: src/components/Faucet/Faq.tsx [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+            </Text>
+            <Text as="p" className={styles.accordionContainer}>
+              A faucet is a platform that gives you test tokens to use when
+              testing your smart contracts. In this case, our faucet is giving
+              you Sepolia ETH to test deploying smart contracts and sending
+              transactions before deploying your dapp in production on Mainnet.
```
Comment (alexandratran) [inline src/components/Faucet/Faq.tsx:223]:
```suggestion
              I'm new to Web3. What is a faucet?
            </Text>
            <Text as="p" className={styles.accordionContainer}>
              A faucet is a platform that gives you testnet tokens to use when
              testing your smart contracts. This faucet gives
              you Sepolia ETH to test deploying smart contracts and sending
              transactions before deploying your dapp on Mainnet.
```

File: src/components/Faucet/Faq.tsx [hunk 8]
```diff
@@ -0,0 +1,277 @@
...
+              <span>
+                <a
+                  data-testid="faq-sepolia-contact-us"
+                  onClick={handleClickContactUs}
+                  target="_blank"
+                  href="https://support.metamask.io/"
+                >
+                  Contact us
+                </a>{" "}
+                with any issues or questions you have relating the faucet.
```
Comment (alexandratran) [inline src/components/Faucet/Faq.tsx:240]:
```suggestion
                  Contact support
                </a>{" "}
                with any issues or questions you have relating to the faucet.
```

File: src/components/Faucet/Faq.tsx [hunk 9]
```diff
@@ -0,0 +1,277 @@
...
+              <span>
+                Have ideas on how to improve the faucet? Awesome! We’d love to
+                hear them. Submit them{" "}
+                <a
+                  data-testid="faq-sepolia-discord"
+                  onClick={handleClickDiscord}
+                  target="_blank"
+                  href="https://discord.com/invite/consensys"
+                >
+                  here.
```
Comment (alexandratran) [inline src/components/Faucet/Faq.tsx:258]:
```suggestion
                If you have ideas on how to improve the faucet,
                feel free to submit them on{" "}
                <a
                  data-testid="faq-sepolia-discord"
                  onClick={handleClickDiscord}
                  target="_blank"
                  href="https://discord.com/invite/consensys"
                >
                  Discord.
```

File: src/components/Faucet/Faq.tsx [hunk 10]
```diff
@@ -0,0 +1,277 @@
...
+          </Accordion>
+          <Accordion>
+            <Text as="h3" className={styles.accordionHeader}>
+              Where does Sepolia ETH come from?
+            </Text>
+            <Text as="p" className={styles.accordionContainer}>
+              The Sepolia ETH comes from our partnership with the Ethereum
+              Foundation. We collaborate with them to support the development
+              community by maintaining an always on and reliable faucet
+              enviroment for the community.
```
Comment (alexandratran) [inline src/components/Faucet/Faq.tsx:271]:
```suggestion
              The Sepolia ETH comes from MetaMask's partnership with the Ethereum
              Foundation. We maintain an always-on, reliable faucet
              environment to support the development community.
```

File: src/components/Faucet/Alerts.tsx [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+      <span>
+        There was an issue starting your transaction. Try again in a few
+        minutes. If the problem persists please{" "}
+        <a
+          data-testid="alert-common-contact-us"
+          onClick={handleClickCommonIssue}
+          target="_blank"
+          href="https://support.metamask.io/"
+        >
+          contact us
```
Comment (alexandratran) [inline src/components/Faucet/Alerts.tsx:62]:
```suggestion
          contact support
```

File: src/components/Faucet/Alerts.tsx [hunk 12]
```diff
@@ -0,0 +1,140 @@
...
+  </div>
+);
+
+export const AlertPastActivity = () => (
+  <div>
+    <AlertTitle>No past activity</AlertTitle>
+    <AlertText>
+      The address used hasn’t been active on Ethereum Mainnet yet. To proceed,
+      please use an address that has activity on Ethereum Mainnet. For further
+      details, refer to our FAQ below.
```
Comment (alexandratran) [inline src/components/Faucet/Alerts.tsx:76]:
```suggestion
      The address used hasn't been active on Ethereum Mainnet yet. To proceed,
      please use an address that has activity on Ethereum Mainnet. For more
      details, refer to the following FAQ.
```

File: src/components/Faucet/Alerts.tsx [hunk 13]
```diff
@@ -0,0 +1,140 @@
...
+        Your current Ethereum address does not contain enough Ether on the
+        Ethereum Mainnet. You can easily add funds to your address using{" "}
+        <a
+          data-testid="alert-balance-add-funds"
+          onClick={handleClickBalanceLow}
+          className="underline"
+          href="https://metamask.io/buy-crypto/"
+        >
+          MetaMask
+        </a>
```
Comment (alexandratran) [inline src/components/Faucet/Alerts.tsx:95]:
```suggestion
    <AlertTitle>Balance too low</AlertTitle>
    <AlertText>
      <span>
        Your current Ethereum address does not contain enough ETH on
        Ethereum Mainnet. You can add funds to your address using{" "}
        <a
          data-testid="alert-balance-add-funds"
          onClick={handleClickBalanceLow}
          className="underline"
          href="https://metamask.io/buy-crypto/"
        >
          MetaMask
        </a>
        .
```

File: src/components/Faucet/Alerts.tsx [hunk 14]
```diff
@@ -0,0 +1,140 @@
...
+  </div>
+);
+
+export const AlertSuccess = ({ url }: { url: string }) => (
+  <div>
+    <AlertTitle>Transaction successfull!</AlertTitle>
+    <AlertText>
+      <span>
+        Your transaction has been sent to the Ethereum/Sepolia network. You
+        should be receiving your ETH shortly.{" "}
```
Comment (alexandratran) [inline src/components/Faucet/Alerts.tsx:127]:
```suggestion
        will receive your ETH shortly.{" "}
```

File: src/components/Faucet/Hero.tsx [hunk 15]
```diff
@@ -0,0 +1,203 @@
...
+              disabled={isLoading}
+              value={inputValue}
+              placeholder="ex. 0x"
+              onChange={handleOnInputChange}
+            />
+            {isLimitedUserPlan && (
+              <p className={styles.caption}>
+                The amount of {network === "linea" && "Linea Sepolia"}
+                {network === "sepolia" && "Sepolia"} ETH you’ll get is
+                determined by your addresses Ethereum Mainnet activity to ensure
```
Comment (alexandratran) [inline src/components/Faucet/Hero.tsx:131]:
```suggestion
                {network === "sepolia" && "Sepolia"} ETH you'll get is
                determined by your address's Ethereum Mainnet activity to ensure
```

File: src/components/Faucet/Maintenance.tsx [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+  };
+
+  return (
+    <div className={styles.maintenance}>
+      <div className={styles.modal}>
+        <MaintenanceIco />
+        <Text as="h3">Our faucet is at full capacity due to high demand</Text>
+        <Text as="p">
+          We’re thrilled by the enthusiasm and are working hard to scale up. Try
+          checking back later. Thanks for your patience. Need ETH urgently?
```
Comment (alexandratran) [inline src/components/Faucet/Maintenance.tsx:32]:
```suggestion
        <Text as="h3">The faucet is at full capacity due to high demand.</Text>
        <Text as="p">
          Try
          checking back later. Thank you for your patience. Need ETH urgently?
```

File: src/components/Faucet/Faq.tsx [hunk 17]
```diff
@@ -0,0 +1,276 @@
...
+                Sepolia ETH?
+              </Text>
+              <Text as="p" className={styles.accordionContainer}>
+                MetaMask requires an address with Ethereum Mainnet activity to
+                safeguard the faucet from automated bots, ensuring equitable
+                Sepolia ETH distribution. The amount of Sepolia ETH you can
+                receive ranges from 0 to 0.5, depending on your address's level
+                of activity. No activity results in no Sepolia ETH, while a
+                higher number of transactions can earn you up to 0.5. MetaMask
+                maintain confidentiality over the exact criteria used to
```
Comment (alexandratran) [inline src/components/Faucet/Faq.tsx:208]:
```suggestion
                maintains confidentiality over the exact criteria used to
```

**Comment by alexandratran** [comment]
I added a bunch of wording edits to better match our documentation style.

Additionally, the faucet works for me, but this is unexpected because I used an address that has had no activity on Ethereum Mainnet ([0x2D48e6f5Ae053e4E918d2be53570961D880905F2](https://etherscan.io/address/0x2d48e6f5ae053e4e918d2be53570961d880905f2)). Additionally, the transaction history shows that the the transactions are still "pending" even though they went through successfully:
Screenshot 2024-10-04 at 3 14 26 PM

**Review by alexandratran — APPROVED** [review]

**Review by aednikanov — APPROVED** [review]

**Review by TrofimovAnton85 — APPROVED** [review]

**Review by cedricmagne — APPROVED** [review]
Works well	
Screenshot 2024-10-09 at 15 27 04

**Pull request merged**